perllib/Integrations/Whitespace.pm

```diff
@@ -125,11 +124,43 @@ sub GetSiteServiceItemRoundSchedules {
 }
 
 sub GetSiteWorksheets {
-    my ($self, $uprn) = @_;
+    my ( $self, $uprn ) = @_;
 
-    my $res = $self->call('GetSiteWorksheets', worksheetInput => ixhash( Uprn => $uprn ));
+    my $res = $self->call( 'GetSiteWorksheets',
+        worksheetInput => ixhash( Uprn => $uprn ) );
 
-    return $res;
+    my $worksheets = force_arrayref( $res->{Worksheets}, 'Worksheet' );
+
+    return $worksheets;
+}
+
+# Needed to get ServiceItemIDs
+sub GetWorksheetDetailServiceItems {
+    my ( $self, $worksheet_id ) = @_;
+
+    my $res = $self->call( 'GetWorksheetDetailServiceItems',
+        worksheetDetailServiceItemsInput =>
+            ixhash( WorksheetId => $worksheet_id ) );
+
+    my $items = force_arrayref( $res->{Worksheetserviceitems},
+        'WorksheetServiceItem' );
+
+    return $items;
+}
+
+sub GetCollectionByUprnAndDate {
+    my ( $self, $uprn, $date_from ) = @_;
+
+    my $res = $self->call(
+        'GetCollectionByUprnAndDate',
+        getCollectionByUprnAndDateInput => ixhash(
+            Uprn                   => $uprn,
+            NextCollectionFromDate => $date_from,
+        ),
+    );
+
+    # TODO Need to force arrayref?
```

Comment:
Will there ever be more than one?

Reply:
Yes, Whitespace tends to send back an array when there's more than one, but just the single item when there's only one, so probably worth forcing it here, have changed that in the fixup commit.

Reply:
You've changed the Plus function but not this one, if that matters :)